### PR TITLE
Various hacks for container-based trusty latest image update

### PR DESCRIPTION
**Pull request by meatballhat**
- patch redis init script to omit `ulimit` call

- [x] back out temporary stuff

### CODE REVIEW

File: lib/travis/build/appliances/fix_container_based_trusty.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        end
+
+        def sh_is_trusty?
+          '$(lsb_release -sc 2>/dev/null) == trusty'
+        end
+
+        def patch_redis_init
+          sh.if "-f #{REDIS_INIT} && sudo grep -q ulimit #{REDIS_INIT}" do
+            sh.raw(
+              %(sudo sed -i -e 's/ulimit/echo HACKED no ulimit/g' #{REDIS_INIT})
```
Comment (cotsog) [inline lib/travis/build/appliances/fix_container_based_trusty.rb]: I would change what we `echo` to not say `HACKED`. Maybe just `no ulimit`?

File: lib/travis/build/appliances/fix_container_based_trusty.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        def sh_is_linux?
+          '$(uname) == Linux'
+        end
+
+        def sh_is_trusty?
+          '$(lsb_release -sc 2>/dev/null) == trusty'
+        end
+
+        def patch_redis_init
+          sh.if "-f #{REDIS_INIT} && sudo grep -q ulimit #{REDIS_INIT}" do
```
Comment (joepvd) [inline lib/travis/build/appliances/fix_container_based_trusty.rb]: The replacement string would also match this test. Not sure if that is a problem longer term?

File: lib/travis/build/appliances/fix_container_based_trusty.rb (hunk 1 above)
Comment (meatballhat) [inline lib/travis/build/appliances/fix_container_based_trusty.rb]: 👍 :trollface:

File: lib/travis/build/appliances/fix_container_based_trusty.rb (hunk 2 above)
Comment (meatballhat) [inline lib/travis/build/appliances/fix_container_based_trusty.rb]: It's definitely sloppy.  /me fix

**Comment by meatballhat** [comment]
Workin' https://staging.travis-ci.org/meatballhat/yolo-octo-adventure/builds/595252

**Pull request merged**